tst/EnergyPlus/api/test_SysExit.py

```diff
@@ -0,0 +1,116 @@
+#     trademark, logo, or other designation of "EnergyPlus", "E+", "e+" or
+#     confusingly similar designation, without the U.S. Department of Energy's
+#     prior written consent.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import shlex
+import tempfile
+import unittest
+
+from pyenergyplus.api import EnergyPlusAPI
+
+
+class TestAPISysExit(unittest.TestCase):
+    """
+    unittest on EnergyPlusAPI to see the exit and flushing to eplusout.err
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        A class method called before tests in an individual class are run
+        """
+        cls.api = EnergyPlusAPI()
+
+    def setUp(self):
+        """
+        Prepare each test fixture, with a new state
+        """
+        self.state = self.api.state_manager.new_state()
+
+    def test_wrong_file_sys_exit(self):
+        """
+        Test for #8483 - system exit called when energyplus can't run
+        """
+        with tempfile.TemporaryDirectory() as tempdir:
+            cmd_args = shlex.split(f'-d {tempdir} -w empty.epw doesntexist.idf')
+            return_code = self.api.runtime.run_energyplus(self.state, cmd_args)
+            self.assertEqual(return_code, 1)
+
+    def test_version(self):
+        """
+        Related #8483 - When we call ProcessArgssystem exit called when energyplus can't run
+        """
+        cmd_args = shlex.split('-v')
+
+        def message_handler(message: bytes) -> None:
+            """
+            Asserts energyplus doesn't try to run
+            """
+            self.assertNotIn("EnergyPlus Starting", message.decode('utf-8'))
+
+        def error_handler(severity: int, message: bytes) -> None:
+            """
+            Same, we shouldn't run, but this checks stderr instead
+            """
+            self.assertNotIn("Input file path", message.decode('utf-8'))
+
+        self.api.runtime.callback_message(self.state, message_handler)
+        self.api.functional.callback_error(self.state, error_handler)
+        return_code = self.api.runtime.run_energyplus(self.state, cmd_args)
+        # Finally, asserts that the return code indicates success
+        self.assertEqual(return_code, 0)
```

Comment:
At bd24465171055ef25113f2ec87be2ab2f2cea433, test fails: **it does try to run the simulation which is something we actually do not want**. it should return after showing the `help`

```
89: Test command: /home/julien/Virtualenvs/py38/bin/python3.8 "-m" "unittest" "/home/julien/Software/Others/EnergyPlus-build-release/Products/test_SysExit.py"
89: Test timeout computed to be: 10000000
89: EnergyPlus, Version 9.5.0-bd24465171
89: Traceback (most recent call last):
89:   File "_ctypes/callbacks.c", line 237, in 'calling callback function'
89:   File "/home/julien/Software/Others/EnergyPlus-build-release/Products/test_SysExit.py", line 100, in message_handler
89:     self.assertNotIn("EnergyPlus Starting", message.decode('utf-8'))
89:   File "/home/julien/.pyenv/versions/3.8.3/lib/python3.8/unittest/case.py", line 1186, in assertNotIn
89:     self.fail(self._formatMessage(msg, standardMsg))
89:   File "/home/julien/.pyenv/versions/3.8.3/lib/python3.8/unittest/case.py", line 753, in fail
89:     raise self.failureException(msg)
89: AssertionError: 'EnergyPlus Starting' unexpectedly found in 'EnergyPlus Starting'
89: EnergyPlus Starting
89: EnergyPlus, Version 9.5.0-bd24465171, YMD=2021.03.01 11:23
89: **FATAL:Input file path  not found
89: Traceback (most recent call last):
89:   File "_ctypes/callbacks.c", line 237, in 'calling callback function'
89:   File "/home/julien/Software/Others/EnergyPlus-build-release/Products/test_SysExit.py", line 106, in error_handler
89:     self.assertNotIn("Input file path", message.decode('utf-8'))
89:   File "/home/julien/.pyenv/versions/3.8.3/lib/python3.8/unittest/case.py", line 1186, in assertNotIn
89:     self.fail(self._formatMessage(msg, standardMsg))
89:   File "/home/julien/.pyenv/versions/3.8.3/lib/python3.8/unittest/case.py", line 753, in fail
89:     raise self.failureException(msg)
89: AssertionError: 'Input file path' unexpectedly found in 'Input file path  not found'
89: EnergyPlus Run Time=00hr 00min  0.09sec
89: Program terminated: EnergyPlus Terminated--Error(s) Detected.
89: FERROR: Could not find input data file: /home/julien/Software/Others/EnergyPlus-build-release/Products/doesntexist.idf.
89: Type 'energyplus --help' for usage.
89: .
89: ======================================================================
89: FAIL: test_version (test_SysExit.TestAPISysExit)
89: Related #8483 - When we call ProcessArgssystem exit called when energyplus can't run
89: ----------------------------------------------------------------------
89: Traceback (most recent call last):
89:   File "/home/julien/Software/Others/EnergyPlus-build-release/Products/test_SysExit.py", line 112, in test_version
89:     self.assertEqual(return_code, 0)
89: AssertionError: 1 != 0
89: 
89: ----------------------------------------------------------------------
89: Ran 2 tests in 0.107s
```

at 9c2ac698fb106798368cb7c717582c79e022706d it passes